src/encoder.rs

```diff
@@ -38,24 +38,26 @@ pub struct Frame {
     pub planes: [Plane; 3]
 }
 
+const FRAME_MARGIN: usize = 16 + SUBPEL_FILTER_SIZE;
```

Comment:
I guessed that restoring of original frame marine would be related to the speed of encoding? Since it enables SIMD code an be used w/o checking the borders. If not, I think I am missing the idea behind how these padded pixels are used

Reply:
yes, padded pixels are used to avoid doing any boundary checks during motion compensation